velox/exec/tests/TableScanTest.cpp

```diff
@@ -39,6 +39,22 @@ using namespace facebook::velox::common::test;
 using namespace facebook::velox::exec::test;
 
 namespace {
+std::string makeCastSql(const variant& v, const TypePtr& type) {
+  std::ostringstream out;
+  out << "CAST('";
+  if (type->isDate()) {
+    // Date variant should be created from string.
+    v.checkIsKind(TypeKind::VARCHAR);
+    out << v.value<TypeKind::VARCHAR>();
```

Comment:
Does v.toJson(DATE()) not work? CC: @majetideepak @aditi-pandit

Reply:
Hi @mbasmanova, the conversion adds extra `"` at the start and end of a date string.
https://github.com/facebookincubator/velox/blob/fe52e60a09e6468d2886c195fef70d99cf5cdce6/velox/type/Variant.cpp#L285
Below error occurs if casting the converted string as date.
> Reason: Unable to parse date value: ""2023-10-27"".Valid date string pattern is (YYYY-MM-DD), and can be prefixed with [+-]

I removed the `"` in this PR, but I'm not sure about it. If it is required, maybe I need to specially handle date type in the test method. Thanks.

Reply:
Thanks for your pointer. I tried to use json function in DuckDB SQL but got below error, so I changed to use `folly::parseJson` to process the json string.

> Reason: DuckDB query failed: Catalog Error: Function with name "json" is not in the catalog, but it exists in the json extension.
> To install and load the extension, run:
INSTALL json;
LOAD json;

Another point is json parsing does not apply to all types, e.g. parsing from below json string gives unexpected result, so I need to handle date specially.

> duckdb> select json('123456789123456789.123456789123456789');
┌───────────────────────────────────────────────┐
│ json('123456789123456789.123456789123456789') │
╞═══════════════════════════════════════════════╡
│ 123456789123456780.0                          │
└───────────────────────────────────────────────┘